homeassistant/components/alexa/smart_home.py

```diff
@@ -0,0 +1,182 @@
+
+    for domain, class_data in MAPPING_COMPONENT.items():
+        for entity in hass.states.async_all():
+            # filter domain
+            if entity.domain != domain:
+                continue
+
+            appliance = {
+                'actions': [],
+                'applianceTypes': [class_data[0]],
+                'additionalApplianceDetails': {},
+                'applianceId': entity.entity_id.replace('.', '#'),
+                'friendlyDescription': '',
+                'friendlyName': entity.attributes.get(ATTR_FRIENDLY_NAME, ''),
+                'isReachable': True,
+                'manufacturerName': 'Unknown',
+                'modelName': 'Unknown',
+                'version': 'Unknown',
+            }
+
+            # static actions
+            if class_data[1]:
+                appliance['actions'].extend(list(class_data[1]))
+
+            # dynamic actions
+            if class_data[2]:
+                supported = entity.attributes.get(ATTR_SUPPORTED_FEATURES, 0)
+                for feature, action_name in class_data[2].items():
+                    if feature & supported > 0:
+                        appliance['actions'].append(action_name)
+
+            discovered_appliances.append(appliance)
+
+    return api_message(
+        'DiscoverAppliancesResponse', 'Alexa.ConnectedHome.Discovery',
+        payload={'discoveredAppliances': discovered_appliances})
+
+
+def extract_entity(funct):
+    """Decorator for extract entity object from request."""
+    @asyncio.coroutine
+    def async_api_entity_wrapper(hass, request):
+        """Process a turn on request."""
+        entity_id = \
+            request[ATTR_PAYLOAD]['appliance']['applianceId'].replace('#', '.')
+
+        # extract state object
+        entity = hass.states.get(entity_id)
+        if not entity:
+            _LOGGER.error("Can't process %s for %s",
+                          request[ATTR_HEADER][ATTR_NAME], entity_id)
+            return api_error(request)
+
+        return (yield from funct(hass, request, entity))
+
+    return async_api_entity_wrapper
+
+
+@extract_entity
+@asyncio.coroutine
+def async_api_turn_on(hass, request, entity):
+    """Process a turn on request."""
+    yield from hass.services.async_call(entity.domain, SERVICE_TURN_ON, {
+        ATTR_ENTITY_ID: entity.entity_id
+    }, blocking=True)
+
+    return api_message('TurnOnConfirmation', 'Alexa.ConnectedHome.Control')
+
+
+@extract_entity
+@asyncio.coroutine
+def async_api_turn_off(hass, request, entity):
+    """Process a turn off request."""
+    yield from hass.services.async_call(entity.domain, SERVICE_TURN_OFF, {
+        ATTR_ENTITY_ID: entity.entity_id
+    }, blocking=True)
+
+    return api_message('TurnOffConfirmation', 'Alexa.ConnectedHome.Control')
+
+
+@extract_entity
+@asyncio.coroutine
+def async_api_set_brightness(hass, request, entity):
+    """Process a set percentage request."""
+    brightness = request[ATTR_PAYLOAD]['percentageState']['value']
```

Comment:
Since this is light specific, let's wrap this call with an `if entity.domain == light.DOMAIN`